dts/arm/nuvoton/npcx9m7fb.dtsi

```diff
@@ -9,16 +9,16 @@
 
 / {
 	flash0: flash@10070000 {
-		reg = <0x10070000 DT_SIZE_K(320)>;
+		reg = <0x10070000 DT_SIZE_K(256)>;
```

Comment:
Wouldn't it make more sense for this change to live in the application? If the hardware has 320 then t he upstream file should indeed say 320, then the application can override it in a leaf file and halve it, don't see any particular problem with it.

Reply:
There are 384 KB RAM in the npcx9m7fb device.
The default sizes of CODE RAM and DATA RAM are 320 KB and 64 KB as described in the datasheet.
However, the size of them is adjustable.
Because the RAM size that could be used by CODE is also limited by the internal flash size (512KB/2), I'd like to change the default to 256KB and 128 KB.
Of course, the application can overwrite it and this commit can be removed from this PR.
We are open to either way that Google prefers to.

Reply:
I think both options are valid, in fairness there was always a mismatch between the full flash and code ram so this is probably fine, let's go with it.